File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.java
class Solution {

    private void helper(int[] dp, int[] cost){
        dp[0] = cost[0];
        dp[1] = cost[1];

        for(int i=2; i<cost.length; i++){
            dp[i] =  cost[i] + Math.min( dp[i-1] ,dp[i-2]);
        }
    }
    public int minCostClimbingStairs(int[] cost) {
        int n = cost.length;
        int[] dp = new int[n];
        helper(dp, cost);
        return Math.min(dp[n-2], dp[n-1]);
    }
}

File: Data Structures & Algorithms/coin-change/submission-0.js
class Solution {
    /**
     * @param {number[]} coins
     * @param {number} amount
     * @return {number}
     */
    coinChange(coins, amount) {
    let res = new Array(amount + 1).fill(amount + 1);
    res[0] = 0;
    
    for (let i = 1; i <= amount; i++) {
        for (let c of coins) {
            if (i - c >= 0) {
                res[i] = Math.min(res[i], res[i - c] + 1);
            }
        }
    }
    
    return res[amount] === amount + 1 ? -1 : res[amount];
    }
}
